CMakeLists.txt

```diff
@@ -36,7 +36,8 @@ if(NOT SWIFT_SYSTEM_NAME)
 endif()
 
 # Don't enable WMO on Windows due to linker failures
-if(NOT CMAKE_SYSTEM_NAME STREQUAL Windows)
+# Don't enable WMO on Android as swift's old driver has a bug with using it.
+if((NOT CMAKE_SYSTEM_NAME STREQUAL Windows) AND (NOT CMAKE_SYSTEM_NAME STREQUAL Android))
```

Comment:
I don't think that this is a good "test". If I am building on Darwin, this should still be enabled yes? Perhaps we should be doing:

```cmake
if(NOT CMAKE_HOST_SYSTEM_NAME STREQUAL Windows)
```

Reply:
I don't know what the windows linker error is so I can't comment on whether there's a host/target windows system dependency or not. The android issue is host specific though so we can use the HOST check for sure

Reply:
although it doesn't really make sense if windows is host/target since we're always going to be building for windows target on windows host. So yeah, actually your suggestion makes sense!